### DISCUSSION TITLE
Unexpected behavior

### DISCUSSION THREAD

**Discussion by deid00** [skhatiri/UAV-Testing-Competition#14] [Platform]
We identify some unexpected behaviour when running some tests.

For instance, the attached test refers to the mission2 case study. From the plot, it seems that the drone passes over the obstacle, but this is not the case as the obstacle height is 15m. Re-running the test without headless mode, we noted that the positions of the obstacles do not match those depicted in the plot. So, the trajectory depicted is (almost) correct, but the blocks are shifted. Are we missing something?

We also observed that the plot is mirrored on the y-axis with respect to the gazebo simulation, but this should not be a problem.

![test_3](https://github.com/user-attachments/assets/76cc878a-a463-46d7-8e54-ca267e103d67)
[test_3.txt](https://github.com/user-attachments/files/17774665/test_3.txt)

**Comment by skhatiri** [comment]
I think you are facing the same problem as described here: https://github.com/skhatiri/UAV-Testing-Competition/blob/master/docs/Q%26A.md#14-i-still-get-some-results-like-the-image-below-in-which-the-drone-is-passing-through-the-obstacle-while-the-height-of-my-obstacles-are-between-15-and-30-meters 
Can you check if the solution there works for you?

**Reply by deid00** [reply]
Thanks for your answer. I'm checking it. 

To the previous post I have also attached the mission file as test_3.txt because yaml files cannot be attached.

**Comment by deid00** [comment]
After checking it seems that we are managing the Obstacle object in the right way
```python
position = Obstacle.Position(
   x=obst['x'], 
   y=obst['y'], 
   z=0,
   r=obst['rotation'],
)
size = Obstacle.Size(
   l=obst['length'], 
   w=obst['width'], 
   h= config.OBSTACLE_HEIGHT, 
)
obstacle = Obstacle(size, position)
```
where obst is a dictionary. The object obstacle is no longer touched.

**Reply by skhatiri** [reply]
Thanks for checking. 
Can you please check if you get the same plot when executing the above yaml test file directly with Aerialist CLI or plotting it again? 
https://github.com/skhatiri/Aerialist?tab=readme-ov-file#command-line-interface 
Let me know if things are ok there?

**Reply by skhatiri** [reply]
If there is no problem there, there is most probably an issue in your code, probably similar to the above one. Maybe you can point us where in code you are generating test cases.

**Reply by deid00** [reply]
Thanks @skhatiri. I have executed the test directly with Aerialist and it works correctly. Moreover, we have also written a dummy executor where we hardcode the coordinates of the obstacles, and it seems to work correctly, as you can see from the plot.

![test_0](https://github.com/user-attachments/assets/fd9be251-1955-4e1e-a32e-01515165c8b4)

The code is (more or less)
```python
        for _ in range(budget):
            obstacle_generator = ObstacleGenerator()
            obstacles, parameters = obstacle_generator.generate(self.case_study_file)
            list_obstacles = []
            for obst in obstacles:               
                position = Obstacle.Position(
                    x=obst['x'], 
                    y=obst['y'], 
                    z=0,
                    r=obst['rotation'],
                )
                size = Obstacle.Size(
                    l=obst['length'], 
                    w=obst['width'], 
                    h= config.OBSTACLE_HEIGHT, # Fixed height of the obstacle
                )
                obstacle = Obstacle(size, position)
                list_obstacles.append(obstacle)
            test = TestCase(self.case_study, list_obstacles)
            try:
                test.execute()
````
However, we are updating our pipeline, so, I let you know if we identify other unexpected case.
